functions/tab.xfce4_terminal.fish

```diff
@@ -0,0 +1,3 @@
+function tab.xfce4_terminal -a cdto cmd
+  xfce4-terminal --tab --working-directory "$cdto" -e "fish -c '$cmd; exec fish'"
```

Comment:
It's functional / useful as-is, so I vote to merge. I'll keep poking at it to see if the history and process control issues can be solved, but that's icing on the proverbial cake.